tests/server.py

```diff
@@ -121,7 +122,15 @@ def request(self, request: Request) -> None:
 
     @property
     def json_body(self) -> JsonDict:
-        return json.loads(self.text_body)
+        body = json.loads(self.text_body)
+        assert isinstance(body, dict)
+        return body
+
+    @property
+    def json_list(self) -> List[JsonDict]:
```

Comment:
I suppose requests could return a list of e.g. strings and ints rather than a list of dicts. But for test code I think this is fine (the tests themselves would hopefully pick up any unexpected type errors).

Reply:
Definitely, but it seemed like an easy small improvement to make. mypy being unable to express JSON sanely gets in here too.